torch/quantization/convert_modules.py

```diff
@@ -36,26 +58,48 @@ def add_observer(module, qconfig_dict, qconfig_parent=None, prefix=''):
         prefix: corresponding prefix of the current module, used as key in
                 qconfig_dict
     """
-    assert not hasattr(module, 'qconfig'), 'Original module should not have \
-        qconfig object already'
-    module.qconfig = None
-    if prefix in qconfig_dict:
-        module.qconfig = qconfig_dict[prefix]
-    else:
-        module.qconfig = qconfig_parent
-
     # Insert observers only for leaf nodes, note that this observer is for
     # the output of the module, for input QuantStub will observe them
     if module.qconfig is not None and len(module._modules) == 0:
         # observer and hook will be gone after we swap the module
         module.add_module('observer', module.qconfig.activation())
         module.register_forward_hook(_observer_forward_hook)
-        return QuantWrapper(module, module.qconfig)
 
     for name, child in module.named_children():
         if name != 'observer':
-            module_prefix = prefix + '.' + name if prefix else name
-            module._modules[name] = add_observer(child, qconfig_dict, module.qconfig, module_prefix)
+            add_observer(child)
+
+def get_config_key(name, qconfig_dict):
+    if name in qconfig_dict:
+        return name
+    elif name:
+        parent = '.'.join(name.split('.')[:-1])
+        return get_config_key(parent, qconfig_dict)
+    else:
+        return '' if '' in qconfig_dict else None
+
+class QuantWrapper(nn.Module):
+    def __init__(self, module, qconfig):
+        super(QuantWrapper, self).__init__()
+        self.quant = QuantStub(qconfig)
+        self.dequant = DeQuantStub()
+        self.module = module
+
+    def forward(self, X):
+        X = self.quant(X)
+        X = self.module(X)
+        return self.dequant(X)
+
+def add_quant_dequant(module):
+    r"""Add QuantStub and DeQuantStub module and Wrap the module in QuantWrapper
+    according to qconfig_dict
+    """
+    is_leaf = len([name for (name, _) in module.named_children() if name != 'observer']) == 0
+    if is_leaf and hasattr(module, 'qconfig') and module.qconfig:
+        return QuantWrapper(module, module.qconfig)
+
+    for name, child in module.named_children():
+        module._modules[name] = add_quant_dequant_helper(child)
```

Comment:
This function is missing?

Reply:
oh, will rename, we don't need the helper function